A.02/Program.cs

```diff
@@ -1,18 +1,59 @@
 ﻿using System;
+using System.Numerics;
+using System.Reflection;
 
-int numberToGuess = new Random().Next(1, 101);
-int UserGuess = 0, tries = 0;
+//Guessing a number generated by computer
+//int numberToGuess = new Random().Next(1, 101);
+//int UserGuess = 0, tries = 0, MaxGuess = 7;
+//bool winner = false;
+
+
+/*while (UserGuess != numberToGuess) {
+    if(tries > MaxGuess - 1) {
+        Console.WriteLine ("You have reached the max number to guesses:" + MaxGuess);
+        break;
+    }
+
+    tries++;
+    Console.WriteLine ("Enter the guess:");
+    bool result = int.TryParse (Console.ReadLine (), out UserGuess);
+    if (result && (UserGuess >= 1 && UserGuess <= 100)) {
+        if (UserGuess < numberToGuess) Console.WriteLine ($"{UserGuess} is Lower");
+        else if (UserGuess > numberToGuess) Console.WriteLine ($"{UserGuess} is higher");
+        else Console.WriteLine ($"{UserGuess} is a correct guess after {tries} tries");
+    } else Console.WriteLine ("Invalid input. Please provide a number between 1 and 100");
+}*/
+
+//Guessing a number thought by user
+
+int lowerLimit = 1, upperLimit = 128;
+var tries=0;
+Console.WriteLine ("Type exact or low or high: ");
+setRange (lowerLimit, upperLimit);
     
-    while (UserGuess != numberToGuess)
-    {
+
+void setRange (int minValue, int maxValue) {
+     bool isExact = false;
+     int guess = midValue (minValue, maxValue);
+
+        while (!isExact) {
         tries++;
-        Console.WriteLine("Enter the guess:");
-        bool result = int.TryParse(Console.ReadLine(), out UserGuess);
-    if (result && (UserGuess>=1 && UserGuess<=100))
-    {
-        if (UserGuess < numberToGuess) Console.WriteLine($"{UserGuess} is Lower");
-        else if (UserGuess > numberToGuess) Console.WriteLine($"{UserGuess} is higher");
-        else Console.WriteLine($"{UserGuess} is a correct guess after {tries} tries");
-    }
-    else Console.WriteLine("Invalid input. Please provide a number between 1 and 100");
-    }    
+     
+            switch (GetResponse (guess)) {
+                //GetResponse (guess).ToUpper;
```

Comment:
Please try to make guesses not like words. Try like 1, 2, 3 options and display these options in each iteration, so user can react fast.